gio/src/win32_input_stream.rs

```diff
@@ -41,12 +41,21 @@ impl fmt::Display for Win32InputStream {
 impl Win32InputStream {
     #[doc(alias = "g_win32_input_stream_new")]
     #[allow(clippy::missing_safety_doc)]
-    pub unsafe fn new<T: IntoRawHandle>(handle: T) -> Win32InputStream {
+    pub unsafe fn take_handle<T: IntoRawHandle>(handle: T) -> Win32InputStream {
```

Comment:
Should this be `T: AsRawHandle`?

Reply:
No, the idea is that `take_handle` takes ownership, while `with_handle` takes a ref. Or am I misunderstanding your comment?

Reply:
Oh, I mean my original question from #381.  For rsvg-convert's stdin/stdout on Windows I want to do something like `Win32InputStream::new(io::stdin())` but that doesn't work, because [`IntoRawHandle`](https://doc.rust-lang.org/std/os/windows/io/trait.IntoRawHandle.html) is only implemented for things like `ChildStdin` and `ChildStdout`.  However, `io::stdin()` returns a `Stdin`, which implements [`AsRawHandle`](https://doc.rust-lang.org/std/os/windows/io/trait.AsRawHandle.html).

I don't really know the Windows API and am just looking at which Rust API bits I can use to get from `io::stdin()/stdout()` to a gio stream.  So, I'm not sure if gtk-rs uses `IntoRawHandle` incorrectly there, when (I think) it should be `AsRawHandle`.  That would make `Win32InputStream::new(io::stdin())` work, since `Stdin` impls `AsRawHandle`.  (`new` or `with_handle` eventually, I think.)

Reply:
To be clear: while the "Into" and "As" prefixes from those traits indeed match the desired behavior of consuming/borrowing, `IntoRawHandle` is not implemented for `Stdin/Stdout` (looks like an oversight in `std`?).

Reply:
I do not know if it is an oversight or a limitation in std.

However Win32InputStream etc can be used for more type of handles beside stdin/stdout, so I think it is correct to have both variants.

For librsvg just use the `with_handle` constructor

Reply:
You are right, `with_handle` should work fine for librsvg.  Thanks!